core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/effects/OpenSearchItem.kt

```diff
@@ -36,7 +36,8 @@ data class OpenSearchItem(
       SearchItemToOpen(searchListItem.value, openInNewTab),
       TAG_FILE_SEARCHED
     )
-    activity.popNavigationBackstack()
+    val readerFragmentResId = (activity as CoreMainActivity).readerFragmentResId
+    activity.navigate(readerFragmentResId)
```

Comment:
Instead of navigating to the new fragment, we should pop the fragment from the stack.